src/SFML/Window/Unix/WindowImplX11.cpp

```diff
@@ -1856,10 +1856,10 @@ bool WindowImplX11::processEvent(XEvent& windowEvent)
                         // There might be more than 1 characters in this event,
                         // so we must iterate it
                         std::uint32_t unicode = 0;
-                        std::uint8_t* iter    = keyBuffer;
-                        while (iter < keyBuffer + length)
+                        std::uint8_t* iter    = keyBuffer.data();
```

Comment:
I think we should use iterators here.

Reply:
Probably wise but I may hold off on that change for now. I think in general we should audit SFML for the use of pointer arithmetic where iterators would be idiomatic.